Tests fail in go 1.14

Changes to go's error messages causing tests to fail.

To reproduce:
```bash
go get golang.org/dl/go1.14
go1.14 download
go1.14 test
```

<details><summary>test output</summary> 
```
Running Suite: libp2p TLS Suite
===============================
Random Seed: 1584021139
Will run 22 of 22 specs

••••Initializing a server using an secp256k1 key: 16Uiu2HAmLvCzPwmReGvF6REB3HArNciMEhDAAozCTxKaswNrth9e
Initializing a client using an RSA key: QmRF7LKTb4mCeK2HqiBCituuoe3frDe9Va2wBh6b2GA5AR

------------------------------
• Failure [1.086 seconds]
Transport
/mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:37
  invalid certificates
  /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:191
    fails if the client presents an invalid cert: private key used in the TLS handshake doesn't match the public key in the cert [It]
    /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:374

    Expected
        <string>: tls: invalid signature by the client certificate: ECDSA verification failure
    to equal
        <string>: tls: invalid certificate signature

    /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:388
------------------------------
Initializing a server using an ECDSA key: QmSwmN69qi5SB1dLKyWUYFH4QmraFtMJsiE4HejJMjMVXf
Initializing a client using an ECDSA key: QmS8fyvTR9HV9JoKt42Sv3EwSYAo1iL8vrgojTnV5TeDtY
• Failure [0.003 seconds]
Transport
/mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:37
  invalid certificates
  /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:191
    fails if the server presents an invalid cert: private key used in the TLS handshake doesn't match the public key in the cert [It]
    /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:404

    Expected
        <string>: tls: invalid signature by the server certificate: ECDSA verification failure
    to equal
        <string>: tls: invalid certificate signature

    /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:424
------------------------------
••Initializing a server using an secp256k1 key: 16Uiu2HAm72ytjXz8dCzP9QWvYQnh1tAhFTDVbxZ2wNzdGCJ3Z8Wb
Initializing a client using an ECDSA key: QmWbWDHGA7zjq29CipEz9GQf1oeGRpk8N3nxw1yNA1YWDn

------------------------------
• Failure [1.018 seconds]
Transport
/mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:37
  invalid certificates
  /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:191
    fails if the client presents an invalid cert: cert is expired [It]
    /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:374

    Expected
        <string>: "...valid: curr..."
    to equal               |
        <string>: "...valid"

    /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:388
------------------------------
Initializing a server using an ECDSA key: QmU8PgQqSvAtVf6Zc1W4AoUoK6ZE7CyRYW9w6R3Rk9PekS
Initializing a client using an ECDSA key: QmdHT9ZjYgvuTcZqU6AwpNY5Js5qga6Wa7N1vsj6oHyqvm
• Failure [0.003 seconds]
Transport
/mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:37
  invalid certificates
  /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:191
    fails if the server presents an invalid cert: cert is expired [It]
    /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:404

    Expected
        <string>: "...valid: curr..."
    to equal               |
        <string>: "...valid"

    /mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:424
------------------------------
••••••••••••

Summarizing 4 Failures:

[Fail] Transport invalid certificates [It] fails if the client presents an invalid cert: private key used in the TLS handshake doesn't match the public key in the cert
/mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:388

[Fail] Transport invalid certificates [It] fails if the server presents an invalid cert: private key used in the TLS handshake doesn't match the public key in the cert
/mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:424

[Fail] Transport invalid certificates [It] fails if the client presents an invalid cert: cert is expired
/mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:388

[Fail] Transport invalid certificates [It] fails if the server presents an invalid cert: cert is expired
/mnt/c/Users/xiege/go/src/github.com/RTradeLtd/go-libp2p-tls/transport_test.go:424

Ran 22 of 22 Specs in 2.778 seconds
FAIL! -- 18 Passed | 4 Failed | 0 Pending | 0 Skipped
--- FAIL: TestLibp2pTLS (2.78s)
FAIL
exit status 1
FAIL    github.com/RTradeLtd/go-libp2p-tls      2.902s
```
</details>

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,7 @@
 language: go
 
 go:
-  - "1.13.x"
+  - "1.14.x"
 
 # first part of the GOARCH workaround
 # setting the GOARCH directly doesn't work, since the value will be overwritten later
@@ -14,7 +14,6 @@ env:
 # now actually set the GOARCH env variable to the value of the temporary variable set earlier
 before_install:
   - |
-    export GO111MODULE=on;
     go get golang.org/x/tools/cmd/cover;
     go get github.com/onsi/ginkgo/ginkgo;
     go get github.com/onsi/gomega;

--- a/go.mod
+++ b/go.mod
@@ -1,6 +1,6 @@
 module github.com/libp2p/go-libp2p-tls
 
-go 1.12
+go 1.14
 
 require (
 	github.com/libp2p/go-libp2p-core v0.3.0

--- a/transport_test.go
+++ b/transport_test.go
@@ -327,9 +327,12 @@ var _ = Describe("Transport", func() {
 
 		transforms := []transform{
 			{
-				name:      "private key used in the TLS handshake doesn't match the public key in the cert",
-				apply:     invalidateCertChain,
-				remoteErr: Equal("tls: invalid certificate signature"),
+				name:  "private key used in the TLS handshake doesn't match the public key in the cert",
+				apply: invalidateCertChain,
+				remoteErr: Or(
+					Equal("tls: invalid signature by the client certificate: ECDSA verification failure"),
+					Equal("tls: invalid signature by the server certificate: ECDSA verification failure"),
+				),
 			},
 			{
 				name:      "certificate chain contains 2 certs",
@@ -339,7 +342,7 @@ var _ = Describe("Transport", func() {
 			{
 				name:      "cert is expired",
 				apply:     expiredCert,
-				remoteErr: Equal("certificate verification failed: x509: certificate has expired or is not yet valid"),
+				remoteErr: ContainSubstring("certificate has expired or is not yet valid"),
 			},
 			{
 				name:      "cert doesn't have the key extension",

